### PR TITLE
Update CMake configuration in preparation for 1.1

### PR DESCRIPTION
**Pull request by lorentey**
This updates the CMake build configuration to incorporate new additions since 1.0.

Of particular note is the new `_CollectionsUtilities` library; it is now imported by every public module of this package.

### Checklist
- [X] I've read the [Contribution Guidelines](/README.md#contributing-to-swift-collections)
- [X] My contributions are licensed under the [Swift license](/LICENSE.txt).
- [X] I've followed the coding style of the rest of the project.
- [ ] I've added tests covering all new code paths my change adds to the project (if appropriate).
- [ ] I've added benchmarks covering new functionality (if appropriate).
- [X] I've verified that my change does not break any existing tests or introduce unexplained benchmark regressions.
- [ ] I've updated the documentation if necessary.

### CODE REVIEW

**Comment by lorentey** [comment]
Cc @compnerd -- I hope I configured the new `_CollectionsUtilities` library correctly.

**Comment by lorentey** [comment]
@swift-ci test

File: Sources/_CollectionsUtilities/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+Copyright (c) 2022 Apple Inc. and the Swift project authors
+Licensed under Apache License v2.0 with Runtime Library Exception
+
+See https://swift.org/LICENSE.txt for license information
+#]]
+
+add_library(_CollectionsUtilities
+  "Array+WithContiguousStorage Compatibility.swift"
+  "CMakeLists.txt"
```
Comment (compnerd) [inline Sources/_CollectionsUtilities/CMakeLists.txt:12]: This should be removed, this is going to cause a build failure.

Comment (lorentey) [reply Sources/_CollectionsUtilities/CMakeLists.txt:12]: Oh good eye -- I wonder how my Ninja test run didn't complain.

Comment (compnerd) [reply Sources/_CollectionsUtilities/CMakeLists.txt:12]: Good question - not sure about that.  There must've been something which prevented that.

Comment (lorentey) [reply Sources/_CollectionsUtilities/CMakeLists.txt:12]: Fixed in https://github.com/apple/swift-collections/pull/203. (Btw, apologies for merging this too soon -- I was trigger happy to get an upcoming merge to `main` go quickly.)

**Pull request merged**